include/benchmark/benchmark_api.h

```diff
@@ -239,6 +242,45 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 // FIXME Add ClobberMemory() for non-gnu compilers
 #endif
 
+
+
+class Counter {
+public:
+
+  typedef enum {
+    kDefaults   = 0x00,
+    // Mark the counter as a rate. It will be presented divided by the duration of the benchmark.
+    kIsRate     = 0x01 << 0,
+    // Mark the counter as a thread-average quantity. It will be presented divided by the number of threads.
+    kAvgThreads = 0x01 << 1
+  } Flags;
+
+public:
+
+  double   value;
+  uint32_t flags;
+
+public:
+
+  BENCHMARK_ALWAYS_INLINE Counter(double v, uint32_t flags_) : value(v), flags(flags_) {}
+  BENCHMARK_ALWAYS_INLINE Counter(double v) : value(v), flags(kDefaults) {}
+  BENCHMARK_ALWAYS_INLINE Counter() : value(0.), flags(kDefaults) {}
+
+  BENCHMARK_ALWAYS_INLINE operator double const& () const { return value; }
+  BENCHMARK_ALWAYS_INLINE operator double      & ()       { return value; }
+
+};
+
+typedef std::map< std::string, Counter > BenchmarkCounters;
+
+// these counter-related functions are hidden to reduce API surface.
+namespace internal {
```

Comment:
If the functions are internal then they shouldn't be declared in the public headers. Please move the declarations elsewhere.